MOBT-801: Using dry equivalents to wet symbols

bayliffe opened the pull request:
Modified wx acceptance test data results once dry equivalents of wet symbols are considered in determining the summary dry symbol.

## Changes
The modifications to the KGO can be seen here:
https://www-nwp/~bayliffe/wxmodal_acceptance_tests/

## Discussed examples

![image](https://github.com/user-attachments/assets/d805618c-4282-4dea-9535-52747887f0ed)

In this example 3 overcast symbols were compared against 6 partly cloudy symbols in the old scheme and the latter chosen as the summary. In the new scheme the rain symbols are dried (given that this is a dry day overall) and contribute to the overcast conditions. This gives a tie between partly cloudy and overcast with the latter chosen to hedge towards the more significant conditions.

---

![image](https://github.com/user-attachments/assets/33a59136-3f1d-439c-826c-3abcd830c699)

This example shows the impact of grouping the cloudy & overcast symbols together, along with the sunny & partly cloudy being grouped together. 5 partly cloudy symbols previously dominated over any other single cloud group, but now the cloudy and overcast are grouped for this comparison and they dominate by 7 to 5. Of these the cloudy symbol is most common, and so it is this which is returned as the summary symbol.

bayliffe commented:
Following conversations this morning, this PR should be considered draft whilst I make some changes to address cloud / clear condition grouping.

Kat-90 commented on data/weather-symbol-modes/intensity_categories.json:
This might also need the fog/mist category adding. If we choose to do this.

brhooper commented on data/weather-symbol-modes/intensity_categories.json:
```diff
@@ -3,5 +3,7 @@
     "snow": [27, 24],
-    "thunder": [30, 29]
+    "thunder": [30, 29],
+    "cloud": [7, 8],
```
Is 'overcast' (8) a more intense option than 'cloudy' (7)? If it is, then should the order of this category be switched? The impact of the current order of this category is shown in the second change in the acceptance test data in your notebook (https://www-nwp/~bayliffe/wxmodal_acceptance_tests/). 

![image](https://github.com/user-attachments/assets/918457d1-6436-405f-9212-4273cd24e803)

bayliffe replied:
See https://github.com/metoppv/improver/pull/2050#discussion_r1860969891

bayliffe commented:
I've now added the visibility group to the config here and updated the test data. The change only impacted the gridded result, and these changes can be seen here:
https://www-nwp/~bayliffe/wxmodal_acceptance_tests_inc_vis/

As with the cloud question, I've hedged to mist rather than fog here such that in a tied situation we get mist. This is because we're bad a vis, so hedging to the more moderate form, but which still shows the likelihood of low vis, seemed prudent.

Kat-90 reviewed (approved):
Thanks for the change, I agree with the mist over fog comment. Maybe we can change this one day!
